mmseg/datasets/custom.py

```diff
@@ -230,6 +244,66 @@ def get_gt_seg_maps(self):
 
         return gt_seg_maps
 
+    def get_classes_and_palette(self, classes=None, palette=None):
+        """Get class names of current dataset.
+
+        Args:
+            classes (Sequence[str] | str | None): If classes is None, use
+                default CLASSES defined by builtin dataset. If classes is a
+                string, take it as a file name. The file contains the name of
+                classes where each line contains one class name. If classes is
+                a tuple or list, override the CLASSES defined by the dataset.
+            palette (Sequence[str] | str | None): If palette is None, use
+                default PALETTE defined by builtin dataset. If palette is a
+                string, take it as a file name. The file contains the name of
+                palette where each line contains one palette value. If palette
+                is a tuple or list, override the PALETTE defined by the
+                dataset.
+        """
+        if classes is None:
+            self.custom_classes = False
+            return self.CLASSES, self.PALETTE
+
+        self.custom_classes = True
+        if isinstance(classes, str):
+            # take it as a file path
+            class_names = mmcv.list_from_file(classes)
+        elif isinstance(classes, (tuple, list)):
+            class_names = classes
+        else:
+            raise ValueError(f'Unsupported type {type(classes)} of classes.')
+
+        if self.CLASSES:
+            if not set(classes).issubset(self.CLASSES):
+                raise ValueError('classes is not a subset of CLASSES.')
+            self.old_id_to_new_id = {}
+            for i, c in enumerate(self.CLASSES):
+                if c not in class_names:
+                    self.old_id_to_new_id[i] = -1
+                else:
+                    self.old_id_to_new_id[i] = classes.index(c)
+
+        return class_names, self.get_palette_for_custom_classes(palette)
+
+    def get_palette_for_custom_classes(self, palette=None):
+
+        if palette:
+            if isinstance(palette, str):
+                # take it as a file path
+                palette = mmcv.list_from_file(palette)
```

Comment:
The `palette` is a list of list. It is not trivial to load palette from path. 
I suggest removing this function. We just need to select the right `palette` if there is.